projects/batfish/src/main/java/org/batfish/grammar/flatjuniper/ConfigurationBuilder.java

```diff
@@ -3292,6 +3319,40 @@ public void exitS_snmp(S_snmpContext ctx) {
   }
 
   @Override
+  public void exitSe_auth_key_chain(Se_auth_key_chainContext ctx) {
```

Comment:
All of this needs to be broken down into multiple rules. Create a context field `_currentAuthenticationKeyChain` that is set upon entry and exit into `se_auth_key_chain`. Then have `exitXXX` rules for the subrules instead of all these `if` statements.